src/utils/object.ts

```diff
@@ -0,0 +1,542 @@
+
+  for (const item of items) {
+    const key = (keyGetter !== null ? keyGetter(item as T) : item) as Key;
+    let group = groups.get(key);
+
+    if (!group) {
+      group = [];
+      groups.set(key, group);
+    }
+
+    group.push(item);
+  }
+
+  return groups;
+}
+
+export function omit<T = unknown>(
+  obj: GenericRecord<T>,
+  keysToOmit: Key | Key[],
+): GenericRecord<T> {
+  keysToOmit = asArray(keysToOmit);
+
+  const keys = Object.keys(obj);
+  const out = {} as GenericRecord<T>;
+
+  for (let i = 0, l = keys.length; i < l; ++i) {
+    const key = keys[i];
+
+    if (!keysToOmit.includes(key)) {
+      out[key] = obj[key];
+    }
+  }
+
+  return out;
+}
+
+export function difference<T = unknown>(arr1: T[], arr2: T[]): T[] {
+  const arr2Set = new Set(arr2);
+  const diff = [];
+
+  for (let i = 0; i < arr1.length; ++i) {
+    const value = arr1[i];
+
+    if (!arr2Set.has(value)) {
+      diff.push(value);
+    }
+  }
+
+  return diff;
+}
+
+export function union<T = unknown>(arr1: T[], arr2: T[]) {
+  if (arr1.length < SMALL_ARRAY_SIZE && arr2.length < SMALL_ARRAY_SIZE) {
+    return unionSmall(arr1, arr2);
+  } else {
+    return unionGeneric(arr1, arr2);
+  }
+}
+
+export function unionSmall<T = unknown>(arr1: T[], arr2: T[]): T[] {
+  const all = arr1.slice();
+
+  for (let i = 0, l = arr2.length; i < l; ++i) {
+    const item = arr2[i];
+
+    if (all.indexOf(item) === -1) {
+      all.push(item);
+    }
+  }
+
+  return all;
+}
+
+export function unionGeneric<T = unknown>(arr1: T[], arr2: T[]): T[] {
+  const all = new Set<T>();
+
+  for (let i = 0; i < arr1.length; ++i) {
+    all.add(arr1[i]);
+  }
+
+  for (let i = 0; i < arr2.length; ++i) {
+    all.add(arr2[i]);
+  }
+
+  return Array.from(all);
+}
+
+export function last<T = unknown>(arr: T[]): T {
+  return arr[arr.length - 1];
+}
+
+export function upperFirst(str: string): string {
+  return str[0].toUpperCase() + str.substring(1);
+}
+
+export function values<T = unknown>(obj: unknown): T[] {
+  if (isObject(obj)) {
+    const rec = obj as GenericRecord<T>;
+    const keys = Object.keys(rec);
+    const values = new Array(keys.length);
+
+    for (let i = 0, l = keys.length; i < l; ++i) {
+      values[i] = rec[keys[i]];
+    }
+
+    return values;
+  } else {
+    return [];
+  }
+}
+
+export function once<T = unknown>(
+  this: any,
+  func: (...args: any[]) => T,
+): (...args: any[]) => T {
+  let called = false;
+  let value: T;
+
+  return (...args: any[]): T => {
+    if (called === false) {
+      called = true;
+      value = func.apply(this, args);
+    }
+
+    return value;
+  };
+}
+
+export function flatten<T = unknown>(arrays: (T | T[])[]): T[] {
+  const out: T[] = [];
+
+  for (let i = 0, l = arrays.length; i < l; ++i) {
+    const value = arrays[i];
+
+    if (Array.isArray(value)) {
+      for (let j = 0; j < value.length; ++j) {
+        out.push(value[j]);
+      }
+    } else {
+      out.push(value);
+    }
+  }
```

Comment:
Can't we just replace it with 
```suggestion
  return arrays.flat(Infinity);
```